File: AndroidMVVMRxJavaRetrofit2/app/src/main/java/com/blackbirds/shakil/androidmvvmrxjavaretrofit2/ViewModel/MainActivityViewModel.java
package com.blackbirds.shakil.androidmvvmrxjavaretrofit2.ViewModel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.blackbirds.shakil.androidmvvmrxjavaretrofit2.Model.PostModel;
import com.blackbirds.shakil.androidmvvmrxjavaretrofit2.Repositroies.MainRepo;

import java.util.List;

public class MainActivityViewModel extends ViewModel {
    private final MainRepo mainRepo;

    public MainActivityViewModel() {
        mainRepo = new MainRepo();
    }

    public LiveData<List<PostModel>> getPostList(){
        return mainRepo.getPostModelLiveData();
    }
}

File: KotlinMVVMRxJavaRetrofit2/app/src/main/java/com/blackbirds/shakil/kotlinmvvmrxjavaretrofit2/ViewModel/MainActivityViewModel.kt
package com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.ViewModel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.Model.PostModel
import com.blackbirds.shakil.kotlinmvvmrxjavaretrofit2.Repositories.MainRepo

class MainActivityViewModel : ViewModel() {

    private val mainRepo: MainRepo

    init {
        mainRepo = MainRepo()
    }

    val getPostList: LiveData<MutableList<PostModel>>
        get() = mainRepo.getPostModelLiveData
}